src/main/java/com/shopping_app/shoppingApp/service/OrderService.java

```diff
@@ -0,0 +1,145 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.BaseException;
+import com.shopping_app.shoppingApp.domain.Address;
+import com.shopping_app.shoppingApp.domain.Cart;
+import com.shopping_app.shoppingApp.domain.CartItem;
+import com.shopping_app.shoppingApp.domain.Order;
+import com.shopping_app.shoppingApp.domain.OrderItems;
+import com.shopping_app.shoppingApp.domain.Product;
+import com.shopping_app.shoppingApp.domain.User;
+import com.shopping_app.shoppingApp.model.Enum.OrderStatus;
+import com.shopping_app.shoppingApp.model.Order.OrderAddRequest;
+import com.shopping_app.shoppingApp.model.Order.OrderResponse;
+import com.shopping_app.shoppingApp.model.Order.OrderUpdateRequest;
+import com.shopping_app.shoppingApp.repository.CartItemRepository;
+import com.shopping_app.shoppingApp.repository.OrderRepository;
+import com.shopping_app.shoppingApp.repository.ProductRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Transactional
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+    private final CartService cartService;
+    private final UserService userService;
+    private final AddressService addressService;
+    private final CartItemRepository cartItemRepository;
+    private final ProductRepository productRepository;
+
+    public List<OrderResponse> getUserOrders(Long userId) {
+        Optional<List<Order>> orders = orderRepository.findAllByUserId(userId);
```

Comment:
why use optional here?

Reply:
Yes its not need here , initially had did some check ( is present and size > 0 ) here, but later i removed which is not neccessary , 
Sure will remove this .

Reply:
updated this